src/esp/physics/bullet/BulletRigidObject.cpp

```diff
@@ -382,14 +334,19 @@ void BulletRigidObject::constructRigidBody(bool kinematic) {
   }
 
   //! Bullet rigid body setup
+  auto motionState = (mt == MotionType::STATIC) ? nullptr : &(btMotionState());
```

Comment:
Taking the address of a temporary object that immediately goes out of scope? This looks shady.

Also, why do you even want to pass a btMotionState here? Is it needed?

Reply:
> Taking the address of a temporary object that immediately goes out of scope? This looks shady.

`
btMotionState& btMotionState() { return *this; }

`
The BulletRigidObject inherits from Magnum::BulletIntegration::MotionState.  This is a way to pass a ref to the BulletRigidObject itself into the btRigidBodyConstructionInfo.

Reply:
Haha, I see now. I guess I've never considered that a method can be the same name as a type. A little confusing to read this line, but anyway that was Magnum's decision in MotionState. @mosra FYI

Reply:
Hah, yeah here it looks very bad :laughing: To deconfuse you could

```suggestion
  auto motionState = (mt == MotionType::STATIC) ? nullptr : &(this->btMotionState());
```